rasa/nlu/components.py

```diff
@@ -754,6 +765,17 @@ def can_handle_language(cls, language: Hashable) -> bool:
         else:
             return language not in not_supported_language_list
 
+    def extract_data_from_domain(self, domain: Domain) -> None:
```

Comment:
Thinking out loud: do we actually need this method to be in the base class aka interface? 

We don't control the `create()` method. Users (and even we ourselves, see `MitieNLP`) overwrite it, so nothing guaranties that `extract_data_from_domain()` will be called (as opposed to other public methods of `Component`, which are called by ROS code). That's why IMO the `extract_data_from_domain()` doesn't belong to the interface.

Also, currently the only implementation says that `Extract the names of all entities from the given domain`, so the better name inside this component would be `extract_entity_names_from_domain`.

I'd rather implement some "helper extract" method only in those implementations that actually need it.

What do you think?